### DISCUSSION TITLE
Using Basti to connect to EFS

### DISCUSSION THREAD

**Discussion by dejan-mladenovski** [BohdanPetryshyn/basti#112] [Q&A] [answered]
Hey @BohdanPetryshyn,

Can I use Basti for accessing EFS volumes? I have this use-case where the app runs in a container (Fargate) and  generates some reports in persistent storage. But, sometimes those reports need to be accessed by some person. Usually they are required for the app runtime, but it can happen that they need to be modified. S3 was my first option but Fargate does not allow mounting S3 as storage. I was thinking if this is possible with Basti to connect directly to the EFS volume, or maybe I need to run an EC2 first, and then Basti to connect to the EC2 instance instead. 

Thanks!

**Comment by BohdanPetryshyn** [comment]
Hi @dejan-mladenovski, thank you for your question!

Additional EC2 instance would definitely allow you to manipulate the EFS from your local machine by connecting to it via SSH with Basti. However, I'll also check if it's possible to mount the EFS on a local machine directly. I'll be able to do this later today or tomorrow.

**Comment by BohdanPetryshyn** [comment, marked as answer]
I was able to connect to EFS from my local machine (Mac) with Basti following this steps:

1. Start port forwarding session with Basti custom connection target
   ```sh
   basti connect \
       --custom-target-vpc <my-vpc-id> \
       --custom-target-host <my-efs-mount-target-ip-address> \
       --custom-target-port 2049 --local-port 2049
   ```

2. Mount the filesystem
   Command suggested in AWS docs didn't work on my machine. After playing with the arguments, I came up with the following command that works.

   ```sh
   mount -t nfs -o vers=4 -o tcp -w localhost:/ Efs
   ```

Please note that the speed of such connection is limited (~300KB/s upload and ~200KB/s download when measured by copying files with `rsync` on my local machine) due to the fact that AWS SSM is used for port forwarding.

**Reply by BohdanPetryshyn** [reply]
Hey @dejan-mladenovski, if this comment answers your question, please mark it as answer 🙏

**Reply by dejan-mladenovski** [reply]
Hey @BohdanPetryshyn I haven't tried it but I trust you and it looks pretty straight forward :) much appreciate the effort!
